src/runtime/schema.ts

```diff
@@ -174,6 +173,9 @@ export class Schema {
         return false;
       }
     }
+    if (Refinement.isAtleastAsSpecificAs(this.refinement, otherSchema.refinement) === AtleastAsSpecific.NO) {
```

Comment:
It seems we were only checking the field level handles previously. @ragavsachdeva can you double check for me that this isn't duplicate work?

Reply:
With the current implementation this is not needed. The system cannot handle multivariate case and will return AtleastAsSpecific.UNKNOWN. In the case where the expression is univariate, it is propagated to field level and is handled there.

However, I added this exact same code in the PR I'm working on where we disable field level refinements. I believe it is better to add this functionality as part of that.